sdk/lib/atl/atlimage.h

```diff
@@ -98,8 +98,8 @@ class CImage
         if (::InterlockedIncrement(&m_nDCRefCount) == 1)
         {
             ATLASSERT(m_hDC == NULL);
-            ATLASSERT(m_hOldBitmap == NULL);
             m_hDC = ::CreateCompatibleDC(NULL);
+            ATLASSERT(m_hOldBitmap == NULL);
```

Comment:
This swapping was not necessary (better have all the checks first).
The other suggestion of Serge to check for the returned values makes sense however.